src/encryption.test.ts

```diff
@@ -5,349 +5,390 @@ import {
+          expect(() =>
+            decrypt({
+              encryptedData: undefined as any,
+              privateKey: bob.ethereumPrivateKey,
+            }),
+          ).toThrow('Missing encryptedData parameter');
+        });
+
+        it('should throw if passed null private key', function () {
+          expect(() =>
+            decrypt({
+              encryptedData,
+              privateKey: null as any,
+            }),
+          ).toThrow('Missing privateKey parameter');
+        });
+
+        it('should throw if passed undefined private key', function () {
+          expect(() =>
+            decrypt({
+              encryptedData,
+              privateKey: undefined as any,
+            }),
+          ).toThrow('Missing privateKey parameter');
+        });
       });
 
-      it('should throw if passed undefined private key', function () {
-        expect(() =>
-          decrypt({
-            encryptedData,
-            privateKey: undefined as any,
-          }),
-        ).toThrow('Missing privateKey parameter');
+      describe('decryptSafely', function () {
+        it('should throw if passed null encrypted data', function () {
+          expect(() =>
+            decryptSafely({
+              encryptedData: null as any,
+              privateKey: bob.ethereumPrivateKey,
+            }),
+          ).toThrow('Missing encryptedData parameter');
+        });
+
+        it('should throw if passed undefined encrypted data', function () {
+          expect(() =>
+            decryptSafely({
+              encryptedData: undefined as any,
+              privateKey: bob.ethereumPrivateKey,
+            }),
+          ).toThrow('Missing encryptedData parameter');
+        });
+
+        it('should throw if passed null private key', function () {
+          expect(() =>
+            decryptSafely({
+              encryptedData,
+              privateKey: null as any,
+            }),
+          ).toThrow('Missing privateKey parameter');
+        });
+
+        it('should throw if passed undefined private key', function () {
+          expect(() =>
+            decryptSafely({
+              encryptedData,
+              privateKey: undefined as any,
+            }),
+          ).toThrow('Missing privateKey parameter');
+        });
       });
     });
+  });
+};
 
-    describe('decryptSafely', function () {
-      it('should throw if passed null encrypted data', function () {
-        expect(() =>
-          decryptSafely({
-            encryptedData: null as any,
-            privateKey: bob.ethereumPrivateKey,
-          }),
-        ).toThrow('Missing encryptedData parameter');
-      });
-
-      it('should throw if passed undefined encrypted data', function () {
-        expect(() =>
-          decryptSafely({
-            encryptedData: undefined as any,
-            privateKey: bob.ethereumPrivateKey,
-          }),
-        ).toThrow('Missing encryptedData parameter');
-      });
+run({
```

Comment:
I'm curious whether it would be obvious that are we are running these tests with different inputs, or even why we are doing so. What are your thoughts on using `describe` to organize/describe the different invocations? I'm thinking an arrangement that looks something like this:

``` typescript
describe('encryption', () => {
  describe('using modern encryption and decryption APIs', () => {
    run({
      decrypt,
      decryptSafely,
      encrypt,
      encryptSafely,
      getEncryptionPublicKey,
    });
  });

  describe('using legacy encryption API', () => {
    run({
      decrypt,
      decryptSafely,
      encrypt: legacyEncrypt,
      encryptSafely: legacyEncryptSafely,
      getEncryptionPublicKey: legacyGetEncryptionPublicKey,
    });
  });

  describe('using legacy decryption API', () => {
    run({
      decrypt: legacyDecrypt,
      decryptSafely: legacyDecryptSafely,
      encrypt,
      encryptSafely,
      getEncryptionPublicKey,
    });
  });
});
```

Note that this would mean removing the `describe('encryption', ...)` from `run`.